site/cds_rdm/inspire_harvester/reader.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 CERN.
+#
+# CDS-RDM is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""Reader component."""
+from urllib.parse import urlencode
+
+import requests
+from datetime import datetime
+from invenio_vocabularies.datastreams.readers import BaseReader
+from invenio_vocabularies.datastreams.errors import ReaderError
+
+
+class InspireHTTPReader(BaseReader):
+    """INSPIRE HTTP Reader."""
+
+    def __init__(self, origin=None, mode="r", since=None, until=None, on=None, inspire_id=None, *args, **kwargs):
```

Comment:
what is the `mode` param?

Reply:
This is the "read" mode for opening the files needed for the parent class `BaseReader`. Actually now that I am looking at it, I don't see the reason for inheriting from `BaseReader`. It just adds additional complexity that we are not benefiting from. I will remove it